src/main/java/net/dv8tion/jda/api/entities/Guild.java

```diff
@@ -2297,6 +2301,68 @@ default Emote getEmoteById(long id)
         return task;
     }
 
+    /**
+     * Retrieves and collects members of this guild into a list.
+     * <br>This will use the configured {@link net.dv8tion.jda.api.utils.MemberCachePolicy MemberCachePolicy}
+     * to decide which members to retain in cache.
+     *
+     * <p><b>This requires the privileged GatewayIntent.GUILD_MEMBERS to be enabled!</b>
+     *
+     * <p><b>You MUST NOT use blocking operations such as {@link Task#get()}!</b>
+     * The response handling happens on the event thread by default.
+     *
+     * @param  roles
+     *         Collection of all roles the members must have
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided
+     * @throws IllegalStateException
+     *         If the {@link GatewayIntent#GUILD_MEMBERS GatewayIntent.GUILD_MEMBERS} is not enabled
+     *
+     * @return {@link Task} - Type: {@link List} of {@link Member}
+     */
+    @Nonnull
+    @CheckReturnValue
+    default Task<List<Member>> findMembersWithRoles(@Nonnull Collection<Role> roles)
+    {
+        Checks.noneNull(roles, "Roles");
```

Comment:
We should check to make sure that the roles given to us are from this guild as a developer sanity check.

We should also consider ensuring that atleast one role was provided.